src/JointLimits.cpp

```diff
@@ -33,4 +33,27 @@ void JointLimits::validate(const samples::Joints& joints) const
     }
 }
 
-} //end namespace base
\ No newline at end of file
+std::pair<bool, samples::Joints> JointLimits::saturate(const samples::Joints& joints)
+{
+    samples::Joints new_joints;
+    new_joints.resize(joints.size());
+    new_joints = joints;
+    bool saturated = false;
+    if (joints.hasNames()) {
+        for (size_t i = 0; i < joints.size(); i++)
+        {
+            auto result = (*this)[joints.names[i]].saturate(joints[i]);
+            saturated = saturated ? saturated : result.first;
```

Comment:
```suggestion
            saturated ||= result.first;
```

Reply:
It's not the same thing!

Reply:
There is no `||=`  operator, but with `|=` these are equivalent: https://godbolt.org/z/73cM6coKr